tests/integration/test_utils/data_source_config/spark_filesystem_csv.py

```diff
@@ -72,7 +113,9 @@ def make_batch(self) -> Batch:
     @override
     def setup(self) -> None:
         file_path = self._base_dir / self.csv_path
-        self.data.to_csv(file_path, index=False)
+        self._spark_data.write.format("csv").option("header", True).options(
+            **self.config.write_options
+        ).save(str(file_path))
```

Comment:
We are writing csv files using spark now. This symmetrical behavior was important because I was having trouble debugging reading datetimes that were written by pandas.

Reply:
It's likely this is actually the required behavior to get the tests passing. There were issues before because the datetimes were being interpreted as strings and could not be compared with lt/gt.